components/eamxx/src/diagnostics/zonal_avg.cpp

```diff
@@ -0,0 +1,186 @@
+#include "diagnostics/zonal_avg.hpp"
+
+#include "share/field/field_utils.hpp"
+
+namespace scream {
+
+void ZonalAvgDiag::compute_zonal_sum(const Field &result,
+  const Field &field, const Field &weight, const Field &lat,
+  const ekat::Comm *comm)
+{
+  auto result_layout = result.get_header().get_identifier().get_layout();
+  const int num_zonal_bins = result_layout.dim(0);
+  const int ncols = field.get_header().get_identifier().get_layout().dim(0);
+  const Real lat_delta = sp(180.0) / num_zonal_bins;
+
+  auto weight_view = weight.get_view<const Real*>();
+  auto lat_view = lat.get_view<const Real *>();
+  using KT = ekat::KokkosTypes<DefaultDevice>;
+  using TeamPolicy = Kokkos::TeamPolicy<Field::device_t::execution_space>;
+  using TeamMember = typename TeamPolicy::member_type;
+  using ESU = ekat::ExeSpaceUtils<typename KT::ExeSpace>;
+  switch(result_layout.rank())
+  {
+    case 1: {
```

Comment:
Do you think we should add support for rank-0 fields as well? If one wants to do a zonal avg of a horiz slice of a 2d quantity, it' be much better to do the slice first, then the zonal avg, since on a 2d field we actually would get coalesced access on GPU in this diagnostic (all other ranks suffer from bad GPU access, but there's not much to do about it).

Reply:
I think I got confused. The output field will ALWAYS have rank>=1, since it has the number of bins as a dimension. Hence, disregard my comment.